### PR TITLE
First pass at adding gitlab CI

### PR DESCRIPTION
**Pull request by adayton1**

### CODE REVIEW

**Comment by adayton1** [comment]
If you haven't already logged on to the CZ gitlab instance, please do so: https://lc.llnl.gov/gitlab. Then I can add you to the mirror of the CARE repo there. Near the bottom of this page: https://lc.llnl.gov/confluence/display/GITLAB/GitLab+CI it sounds like you need to add a comment like "LGTM" to actually trigger the gitlab CI. But I'm not sure if that will happen on this pull request or if it needs to be in develop first.

File: .gitlab/ci/helper.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+######################################################################################
+# Copyright 2020 Lawrence Livermore National Security, LLC and other CARE developers.
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+######################################################################################
+
+mkdir build
```
Comment (dtaller) [inline .gitlab/ci/helper.sh:10]: If possible, I would change this to mkdir build-cuda or mkdir build-blueos-nvcc10.1.243_clang8.0.1.cmake or something more descriptive.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# SPDX-License-Identifier: BSD-3-Clause
+######################################################################################
+
+variables:
+   GIT_SUBMODULE_STRATEGY: normal
+
+stages:
+   - build
+
+gpu:
```
Comment (dtaller) [inline .gitlab-ci.yml:14]: Would prefer this to be cuda or something more specific then gpu, because someday we might also have amd gpus

**Review by dtaller — COMMENTED** [review]
Overall fine, tell me if you think my comments on name changes are too nitpicky or if they are reasonable

File: .gitlab/ci/helper.sh (hunk 1 above)
Comment (adayton1) [reply .gitlab/ci/helper.sh]: I changed it to build-cuda. When we actually have more than one build, I will definitely be more specific.

File: .gitlab-ci.yml (hunk 2 above)
Comment (adayton1) [reply .gitlab-ci.yml]: Good point. I changed the name to cuda.

Comment (adayton1) [reply .gitlab-ci.yml]: I will leave more specific names for when we actually do more than one build.

**Review by dtaller — APPROVED** [review]
Thanks for the name changes. Looks good to me now!

File: .gitlab-ci.yml (hunk 2 above)
Comment (dtaller) [reply .gitlab-ci.yml]: Thanks!

File: .gitlab/ci/helper.sh (hunk 1 above)
Comment (dtaller) [reply .gitlab/ci/helper.sh]: Thanks!

**Review by liu15 — APPROVED** [review]

**Pull request merged**